### ISSUE TITLE
Make it possible to open a ROM by double clicking it in the file explorer

### ISSUE THREAD

**Issue by geajack** [libretro/RetroArch#9275]
I use Ubuntu. I'd like to be able to start a ROM by just finding it in my file system and double clicking it. On Ubuntu, you would normally do this by associating a file type with a given program, so you would associate Retroarch with ".smc" files and then you could open them with a simple double click.

However, Retroarch doesn't show up as an option when I right click on a file and go to "Open with...". This is some sort of issue with the Retroarch installation package for Ubuntu, I think it needs to stick a .desktop file somewhere.

There's a deeper issue, though... Retroarch just doesn't support "opening a ROM" as far as I know. If I could run a ROM by simply going

```
$ retroarch "Super Mario World (U) [!].smc"
```

then I could put together the necessary config files to get Retroarch to show up in the "Open with" menu and solve my problem. It seems like this is a two part issue:

1. Support opening a ROM with the syntax `$ retroarch rom_file`.
2. Fix the Debian installation package (or Snap or whatever it is) to set things up so Retroarch shows up in the Open With menu.

**Comment by RobLoach** [comment]
Would love for this to get in. Similar issue over at https://github.com/libretro/RetroArch/issues/746 .

**Comment by bparker06** [comment]
>Retroarch just doesn't support "opening a ROM"

Yes because it has no idea what core you want to use. How should that issue be solved? Not everyone will always use the same core for all content from the same system/extension either.

**Comment by LazyBumHorse** [comment]
> Yes because it has no idea what core you want to use. How should that issue be solved?

There's an obvious solution:
 - If there's just one core supporting the extension, use that
 - If there's multiple, show the core selection dialog

This is _exactly_ what the file browser already does and it could be implemented by essentially loading up "Load Content" with the given path.
This could even work for directories, so you could start RetroArch and be immediatly able to select and load a ROM.

Currently, at least on Windows, it's also possible to use file extension associations by for example assigning `.sfc` with retroarch.exe, opening Regedit and editing the generated `sfc_auto_file` open command to use e.g. the Snes9x core like this:
```
[HKEY_CLASSES_ROOT\sfc_auto_file\shell\open\command]
@="\"D:\\RetroArch\\retroarch.exe\" -L \"cores\\snes9x_libretro.dll\" \"%1\""
```

**Comment by MarcoEstevez** [comment]
Actually there is another LibRetro frontent who does that

https://github.com/Aftnet/RetriX

**Comment by geajack** [comment]
> > Retroarch just doesn't support "opening a ROM"
> 
> Yes because it has no idea what core you want to use. How should that issue be solved? Not everyone will always use the same core for all content from the same system/extension either.

For other users: see [746](https://github.com/libretro/RetroArch/issues/746) for further discussion.
